### ISSUE TITLE
Add functionality to add custom job titles to landing page sign-up form

### ISSUE THREAD

**Issue by joshdanielson** [loconomics/loconomics#367]

**Comment by IagoSRL** [comment]
@joshdanielson For this to work, do you want in the landing page to hide or lock the sign-up fields until the user clicks on a job title from the list, or the user can sign-up without choose any?

**Comment by joshdanielson** [comment]
@situatedbit and I talked about this a little yesterday. So as of now, the landing page will stay the same no matter what the job titles. The ONLY thing that will change are the job titles that show in the sign-up form. I was thinking we could do some sort of dictionary in the .js file the would read a URL parameter that I could add to the URL, e.g. loconomics.com/welcome/sanfrancisco/healingarts  (or a number)

So it would look something like this:
var jobtitles = {"healingarts": {"Massage Therapist": 106, "Acupuncturist": 185, "Bodyworker": 310}, "petcare": {"Dog Walker": 76, "Dog Sitter": 78, "Cat Sitter": 311};

**Comment by situatedbit** [comment]
I'm going to work a little on the UI here.

**Comment by IagoSRL** [comment]
@situatedbit Don't know how do you see the implementation of this, but I was thinking in the list of job titles be outside the signup template, with a 'selected job title' observable that is read on change and set into the new signup.jobTitleID (added at #331) at the landingPage activity. To don't add more complexity into the signup template plus viemodel. What do you think?

(the code that keep in sync the selected value can be as short as:)
``` jobTitlesListViewModel.selectedJobTitleID.subscribe(signup.jobTitleID) ```

**Comment by situatedbit** [comment]
Yes, I think this is good. I'm proposing a few changes to the sign up form below.

**Comment by situatedbit** [comment]
![job-title-signup](https://cloud.githubusercontent.com/assets/6644016/25500180/e655cff2-2b42-11e7-9185-8510cba98892.png)

**Comment by situatedbit** [comment]
@joshdanielson @IagoSRL Any thoughts about this? ↑

**Comment by IagoSRL** [comment]
I wanted to pick up this today, but ending the other things and details couldn't. Well, only just thinking on some impl details but not coded yet.

The proposed design looks fine to me. I will sleep now, so you have some time to decide at your afternoon, so I can work on it tomorrow morning.

**Comment by IagoSRL** [comment]
Uploaded to *dev* my progress on this, with new design implemented additionally to the job titles list

**Know bug** normal site signup and learnMoreProfessionals are broken now, but landing page works.

Note: from the design, the 'agree terms' language is hidden until 'signup with email' button is clicked, I think must be visible all the time because affects too to sign-ups using Facebook button.

Last: the code to integrate signup component and set a list of job titles, as you can check on this branch at the landing page html, is like this:
```
                        <app-signup params="isCountryVisible: false, onSignedUp: redirect, profile: profile">
                            <h5 class="SignupLogin-prompt">What is your primary area of work?</h5>
                            <app-select-job-title params="selected: jobTitleID, caption: 'Choose one to start'">
                                <item id="106">Massage Therapist</item>
                                <item id="185">Acupuncturist</item>
                                <item id="310">Bodyworker</item>
                                <item id="0">Other</item>
                            </app-select-job-title>
                        </app-signup>
```

**Comment by joshdanielson** [comment]
@IagoSRL nice work! Can you add a caret to that field so it looks like a drop-down? The Country dropdown had the style.

Also, #226 is done from a copy standpoint. I added testimonials today, and it's looking good :)

**Comment by situatedbit** [comment]
Yes, looks good!

One issue: if I skip add job titles during onboarding, the back link on mobile points to calendar: 

![onboarding-calendar](https://cloud.githubusercontent.com/assets/6644016/25551645/a50e5014-2c3c-11e7-9737-424d516070aa.jpg)

**Comment by situatedbit** [comment]
Also, on a phone the scroll-to-element bug is pretty bad. Until we fix #377, can we disable the behavior by defaulting ``hasScrolledToField = true`` in ``viewmodels/ValidatedPassword.js``?

**Comment by situatedbit** [comment]
Unfortunately I see a problem in my design. 

The _then_ and _or_ lines can be misunderstood to mean (choose job title then sign up for facebook) or (sign up with email).

Can we move the horizontal rule style from _or_ to _then_ like in this new version of the mockup? 

![landing-page](https://cloud.githubusercontent.com/assets/6644016/25551742/4bcfe1fa-2c3e-11e7-877a-32e2a3780b9a.png)

**Comment by IagoSRL** [comment]
I fixed the Friday bug on this for web sign-up and learnMore pages and did the requested fixes/changes, including #377 (that I implemented on this branch) and some other minor fixes.
Thanks @joshdanielson for adding the dropdown caret.
@situatedbit I was unable to reproduce the bug with the back link when skipping job title; maybe was a temporary bug?

Merged this branch into #266 and closed

Uploaded latest work at branch is266 at *dev*.
